gslab_scons/record_dir.py

```diff
@@ -0,0 +1,113 @@
+import os
+import scandir
+import mmh3
+
+from misc import check_and_expand_path, make_heading
+
+def record_dir(start_path, name, 
+               include_checksum = False,
+               file_limit = 5000, 
+               logpath = 'state_of_external.log'):
+    '''
+    Record relative path, size, and (optionally) checksum of all files within start_path.
+    Relative paths are from start_path.
+    Append info in |-delimited format to logpath below a heading made from start_path.
+    '''
+    check_logpath(logpath)
```

Comment:
So where can we possibly get a bad path? I see `logpath` and `start_path`. We should see how we want to deal with this. I don't think throwing an error is a good idea, since it's possible that this subdirectories didn't exist prior to running builders / SCons creating all the necessary subdirectories. Maybe writing "we can't find your directory 🤷‍♀️ " to file?

Reply:
We can get bad paths if the user specifies 
* inpath as a directory that doesn't exist
* outpath in a directory that doesn't exist

New implementation does the following
* `inpath`: Check that `inpath` exists. If not, add ', could not find at runtime.' to its name. If `inpath` exists as a file, change `inpath` to the directory and record that we only want information about that file. 
* `outpath`: Check that the dir where we put `outfile` is a directory. If not, try to make it. (If the directory exists as a file, we throw an error.)